filters/filter.go

```diff
@@ -56,11 +56,32 @@ type Filter interface {
 }
 
 // NewFilter supports startsWith, endsWith, contains and a single wildcard
-// TODO(martinm): add rest of glob matching support and negation
+// along with negation
+// TODO(martinm): add rest of glob matching support
 func NewFilter(pattern string) (Filter, error) {
-	idx := strings.Index(pattern, WildcardChar)
+	nIdx := strings.Index(pattern, negationChar)
+	if nIdx != 0 {
```

Comment:
I think we actually drop those illegal characters from metric names today. If you take a look at `sanitize` function in `sanitize.go` under the `filter` package, it's used in both the path filter and the tags filter, which replaces illegal chars with empty string. This means when we get to match the metric names against rules, the metric should have been sanitized and we don't need to worry about the illegal chars.

If that's the case, perhaps we should be consistent and disallow "!" in rules other than the first character (and possibly other illegal characters) as well. Thoughts?